File: 2020/python/05.py
from aocd import data

trans = dict(zip(map(ord, "FLBR"), "0011"))

seats = {int(line.translate(trans), 2) for line in data.split("\n")}

print(max(seats))

for x in seats:
    if x + 1 not in seats and x + 2 in seats:
        print(x + 1)

File: 2020/ruby/05.rb
require './utils.rb'
s = read(5).lines.map{|line|line.tr('FLBR', '0011').to_i 2}

p s.max
p (s.min..s.max).find{|i| s.include?(i-1) && !s.include?(i) && s.include?(i+1)}
